src/main/groovy/com/morpheusdata/xen/XenserverBackupExecutionProvider.groovy

```diff
@@ -101,7 +101,38 @@ class XenserverBackupExecutionProvider implements BackupExecutionProvider {
 	 */
 	@Override
 	ServiceResponse deleteBackupResult(BackupResult backupResult, Map opts) {
-		return ServiceResponse.success()
+		log.debug("Delete backup result {}", backupResult.id)
+		ServiceResponse rtn = ServiceResponse.prepare()
+		try{
+			def snapshotId = backupResult.snapshotId
+			def sourceWorkload = plugin.morpheus.async.workload.get(opts?.containerId ?: backupResult.containerId).blockingGet()
+			ComputeServer computeServer = sourceWorkload.server
+			Cloud cloud = computeServer.cloud
+			//TODO: below lines of code needs core support
+//			if(backup.copyToStore) {
+//				super.cleanBackupResult(backupResult)
+//			}
+			if(cloud) {
+				Map authConfig = plugin.getAuthConfig(cloud)
+				def result = XenComputeUtility.destroyVm(authConfig, snapshotId)
+				log.debug("delete xen snapshot result: {}", result)
+				if(!result?.success) {
+					rtn.success = false
+				} else {
+					def snapshotRecord = computeServer?.snapshots?.find{it.externalId == snapshotId}
+					if(snapshotRecord) {
+						computeServer.snapshots.remove(snapshotRecord)
+						morpheusContext.async.computeServer.save(computeServer).blockingGet()
```

Comment:
The snapshot list may not get modified in the method implemented, you may need to do somehting like:
```
def snapList = computeServer.snapshots
snapList.remove(snapshotRecord)
computeServer.snapshots = snapLIst
```